test/Macros/PointerBounds/CountedBy/CountExpr.swift

```diff
@@ -0,0 +1,21 @@
+// REQUIRES: swift_swift_parser
+
+// RUN: %target-swift-frontend %s -swift-version 5 -module-name main -disable-availability-checking -typecheck -enable-experimental-feature SymbolLinkageMarkers -plugin-path %swift-plugin-dir -dump-macro-expansions > %t/expansions-dump.txt | %FileCheck %s
+
+@PointerBounds(.countedBy(pointer: 1, count: "size * count"))
+func myFunc(_ ptr: UnsafePointer<CInt>, _ size: CInt, _ count: CInt) {
+}
+
+// CHECK-NOT: @PointerBounds
+// CHECK: func myFunc(_ ptr: UnsafePointer<CInt>, _ size: CInt, _ count: CInt) {
+// CHECK-NEXT: }
+// CHECK-NEXT: 
+// CHECK-NEXT: func myFunc(_ ptr: UnsafeBufferPointer<CInt>, _ size: CInt, _ count: CInt) {
+// CHECK-NEXT:     let _ptrCount: some BinaryInteger = size * count
```

Comment:
I'd rather not have `_ptrCount` typed to `some BinaryInteger`, which is erasing information when we could be more precise. Maybe just leave off the type annotation and let the compiler figure it out?

Reply:
I did this because the expression could be any type, and in the case that it does have the wrong type, the type error here makes it clearer that the error is because of the user annotation rather than a bug in the macro. Maybe there's a neater way to do something like this?
```
let _ptrCount = size * count
let _assertPtrCountBinaryIntegerType: some BinaryInteger = _ptrCount // unused, only for asserting type
```

Reply:
I guess if we have a function like this we could form a call to it: 

```swift
@_alwaysEmitIntoClient 
func _assertBinaryInteger<T: BinaryInteger>(_: T) { }
```

... but the result is basically the same, so what you've done will work. My main concern is that the opaque type we're creating here will have some runtime impact that doesn't get optimized away.